src/cr/cube/measures/pairwise_pvalues.py

```diff
@@ -0,0 +1,74 @@
+# encoding: utf-8
+
+"""P-values of pairwise comparison or columns of a contingency table."""
+
+from __future__ import division
+
+import numpy as np
+
+from cr.cube.distributions.wishart import wishartCDF
+from cr.cube.util import lazyproperty
+
+try:
+    xrange
+except NameError:  # pragma: no cover
+    xrange = range
+
+
+class PairwisePvalues(object):
+    """Value object providing matrix of pairwise-comparison P-values"""
+
+    def __init__(self, slice_, axis=0):
+        self.slice = slice_
+        self.axis = axis
+
+    @lazyproperty
+    def values(self):
+        """Square ndarray of pairwise Chi-square along axis."""
+        chisq = self.pairwise_chisq(axis=self.axis)
+        return 1.0 - wishartCDF(chisq, self.n_min, self.n_max)
+
+    @lazyproperty
+    def n_max(self):
+        """zero based int representing the bigger of the two cube's dimension."""
+        return max(self.slice.get_shape()) - 1
+
+    @lazyproperty
+    def n_min(self):
+        """zero based int representing the smaller of the two cube's dimension."""
+        return min(self.slice.get_shape()) - 1
+
+    def pairwise_chisq(self, axis=0, weighted=True):
```

Comment:
Overall I like this class, although I wonder why `axis` and `weighted` can't be made construction-time arguments and make the entire interface lazyproperties. This is the characteristic of a value object; you create it and then you can read its values (no methods).

This decision is made based on expected usage, but a value object is inherently simpler to reason about and easier to understand. I'm strongly inclined to use them whenever possible and I generally find it possible. In the end the question is what the calling protocol will need to look like in the calling code, which is this class's reason for being.